qml/Shell.qml

```diff
@@ -552,7 +552,7 @@ StyledItem {
             }
 
             readonly property bool focusedSurfaceIsFullscreen: topLevelSurfaceList.focusedWindow
-                ? topLevelSurfaceList.focusedWindow.state === Mir.FullscreenState
+                ? topLevelSurfaceList.focusedWindow.state == Mir.FullscreenState
```

Comment:
This looks very suspicious; if it works with the new code but not with the old, it looks like we might have some error in registering the enums.

Reply:
I agree there's no reason, but the bug is architecture-specific. Strict comparison worked on amd64 but not armhf or arm64. The values being returned are correct but for whatever reason they don't have the right type.

Reply:
Could you please file a bug about this, so when someone has some time he can investigate this issue?